Review 56 clip api for layer depths

jovannyrodz opened the pull request:
Adding dynamic ability to clips to move depths within a layer
Adding dynamic ability to clips to change from a layer into another.

Alani25 reviewed (changes requested):
Tested both in preview, preview window, and HTML/ ZIP exports, and code works as expected. 
I found the swapDepth function interesting in particular, it looks pretty useful.
There is one thing I'd like you to do, and that is to add these new functions with a short description under the code reference just so we have it documented somewhere the user can find and use easily.

<img width="151" height="360" alt="image" src="https://github.com/user-attachments/assets/6dc6b28e-9fe5-4d42-a0ee-f2244e90d142" />

Also while you're at it, either in this branch or another, add your sound code as well.
I believe the code for the reference is inside `src` >> `editor` >> `scriptInfo.js` and hopefully easy to add.

That aside all looks good 
Thanks for your hard work :)

jovannyrodz commented:
> pumpkinhead's name is misspelled (missing a p), and i think the sendForward() and sendBackward() were also made by him. other than this and what hamzah said, this should be all set.

Sure,  I'll fix his name, but I added my versions of sendForward() and sendBackward() functions.   I'll not put my name either.

jovannyrodz commented:
> Tested both in preview, preview window, and HTML/ ZIP exports, and code works as expected. I found the swapDepth function interesting in particular, it looks pretty useful. There is one thing I'd like you to do, and that is to add these new functions with a short description under the code reference just so we have it documented somewhere the user can find and use easily.
> 
> <img alt="image" width="151" height="360" src="https://private-user-images.githubusercontent.com/80855152/483897290-6dc6b28e-9fe5-4d42-a0ee-f2244e90d142.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY1OTMxMTUsIm5iZiI6MTc1NjU5MjgxNSwicGF0aCI6Ii84MDg1NTE1Mi80ODM4OTcyOTAtNmRjNmIyOGUtOWZlNS00ZDQyLWEwZWUtZjIyNDRlOTBkMTQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODMwVDIyMjY1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhYTk1MWNlYzEzMzk5MDJiYTQ5YTUxNjk0YTA0MzBhMDA1ZjM4OTdlOWFmZDAzMTExNDE0NTNlZGJiODY5NTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.B5sekYso2H-omsQ3p1u6l87TGaLrOLncg4LTTob006A">
> Also while you're at it, either in this branch or another, add your sound code as well. I believe the code for the reference is inside `src` >> `editor` >> `scriptInfo.js` and hopefully easy to add.
> 
> That aside all looks good Thanks for your hard work :)

Ok, I just added two more methods and updated the scriptInfo.

BaronAWC commented on engine/src/base/Clip.js:
```diff
@@ -1334,6 +1334,154 @@ let avgIntersection = {
+  /**
+   * depth - Send clip to Absolute Front: top layer, top depth
+   */
+  sentToAbsoluteFront() {
```
should be send, not sent

jovannyrodz replied:
Dang, LOL That was GPT5.

jovannyrodz replied:
Nice catch

BaronAWC commented on src/Editor/scriptInfo.js:
```diff
@@ -244,23 +244,66 @@ class ScriptInfoInterface extends Object {
+                  },
+                  {
+                    name: 'sentToAbsoluteFront',
+                    snippet: 'this.sentToAbsoluteFront()',
```
same thing here, use send instead of sent

BaronAWC reviewed (changes requested):
all good other than renaming

Alani25 reviewed (approved):
Love the two new functions you added! 
Also thanks for adding them to the reference code, there's a lotta new functions they're hard to remember so that helps a lot. The detailed info you added under each makes them easy to understand as well for beginners.

Asides from Baron's notes and quick fixes I don't have much to add.
Great stuff :)

jovannyrodz commented:
> Love the two new functions you added! Also thanks for adding them to the reference code, there's a lotta new functions they're hard to remember so that helps a lot. The detailed info you added under each makes them easy to understand as well for beginners.
> 
> Asides from Baron's notes and quick fixes I don't have much to add. Great stuff :)

That part was done by gpt5 : )
